### PR TITLE
[multibody] Document default collision filtering

### PR DESCRIPTION
**Pull request by rpoyner-tri**
Closes: #13672
Relevant to: #16422

Write down what we do now, add a clarification and cross-reference to the
SceneGraph docs, and delete some obsolete text.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)

### CODE REVIEW

**Review by rpoyner-tri — COMMENTED** [review]
+@amcastro-tri for feature review.
FYI @hongkai-dai , @ericcousineau-tri 
+(release notes: none) 

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490#-:-MurYMpkDKl7A5_U7OSQ:b-fzyw1a)* status: LGTM missing from assignee amcastro-tri, needs platform reviewer assigned, needs at least two assigned reviewers (waiting on @amcastro-tri)

**Review by amcastro-tri — COMMENTED** [review]
minor comments. :lgtm: 

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490#-:-Muztf_65fz7lidLs81H:bf8ttum)* status: 3 unresolved discussions, needs platform reviewer assigned, needs at least two assigned reviewers (waiting on @rpoyner-tri)

___
*[geometry/scene_graph.h, line 809 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Muzq7KE-Dfolat-vzOS:-Muzq7KE-Dfolat-vzOT:b3pvth5) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/geometry/scene_graph.h#L809)):*
> ```objc
>    data (model or context).
> 
>    Note: %SceneGraph does not track topology or semantic information of models,
> ```

fyi, you might rather prefer a Doxygen `@note` instead?
___
*[geometry/scene_graph.h, line 811 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Muzpw2g7boouoRDXBMm:-Muzpw2h6WjHvTyQ4xlh:b-9adzz5) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/geometry/scene_graph.h#L811)):*
> ```objc
>    Note: %SceneGraph does not track topology or semantic information of models,
>    so decisions about *what* to filter belong with
>    drake::multibody::MultibodyPlant or its clients.
> ```

nit, this to me reads as if with "clients" you referred to MBP's clients. Maybe rephrase to:
"... belong to SceneGraph's clients such as drake::multibody::MultibodyPlant."
___
*[multibody/plant/multibody_plant.h, line 512 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-MuzqTb53h60IhdT6hZI:-MuzqTb53h60IhdT6hZJ:b-mz8sbo) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/multibody/plant/multibody_plant.h#L512)):*
> ```objc
> Once the user is done adding modeling elements and registering geometry, a
> call to Finalize() must be performed. This call will:
> - Build the underlying MultibodyTree topology, see MultibodyTree::Finalize()
> ```

nit, I know not part of this PR but MBT is `internal::`. Could we change the first bullet to: "Build the underlying tree structure of the multibody model."
___
*[multibody/plant/multibody_plant.h, line 519 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-MuztGu9APZP7kl6oG2j:-MuztGuACI_IiACuWbMT:byau1a5) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/multibody/plant/multibody_plant.h#L519)):*
> ```objc
> - declare collision filters to ignore collisions:
>   - between bodies connected by a joint,
>   - between bodies welded to the world.
> ```

nit, maybe say: "between bodies welded (directly or transitively) to the world". Not sure if this as is implies that islands of welded bodies to the world are actually filtered or not. They are. Fell free to reword or ignore if clear to you.

**Review by amcastro-tri — COMMENTED** [review]
+@jwnimmer-tri for platform review please.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490#-:-Muzu11M5zV05ft8yuvX:b-qxjhdt)* status: 3 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform) (waiting on @jwnimmer-tri and @rpoyner-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
:lgtm: platform.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490#-:-Mv-DPuN8Lm2hRJPg0ip:b-vtj4vo)* status: 3 unresolved discussions (waiting on @rpoyner-tri)

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: 4 unresolved discussions (waiting on @rpoyner-tri)

___
*[geometry/scene_graph.h, line 809 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Muzq7KE-Dfolat-vzOS:-Mv-DJ5vAanWMyF1KOfi:b-t0w6al) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/geometry/scene_graph.h#L809)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

fyi, you might rather prefer a Doxygen `@note` instead?
</blockquote></details>

good call. Working
___
*[geometry/scene_graph.h, line 811 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Muzpw2g7boouoRDXBMm:-Mv-DMU9C_Y9icRtq3VR:b-nx9qsx) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/geometry/scene_graph.h#L811)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

nit, this to me reads as if with "clients" you referred to MBP's clients. Maybe rephrase to:
"... belong to SceneGraph's clients such as drake::multibody::MultibodyPlant."
</blockquote></details>

This is a good comment -- you are making me think harder. I did originally mean mbp or its clients, thinking really of "its clients" as application code, essentially. My assumption was that SceneGraph doesn't really have any other viable partners/consumers than mbp. I think that assumption is largely true, or at least, I don't know of any significant counterexamples.

But, as @ggould-tri teaches, "if you're reviewer is confused, your work is objectively confusing." Perhaps I'll work on a phrasing that is even less prescriptive about what code can make filtering decisions.
___
*[multibody/plant/multibody_plant.h, line 512 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-MuzqTb53h60IhdT6hZI:-Mv-FzeFCVXT8TNYdM8D:b39o3f2) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/multibody/plant/multibody_plant.h#L512)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

nit, I know not part of this PR but MBT is `internal::`. Could we change the first bullet to: "Build the underlying tree structure of the multibody model."
</blockquote></details>

Will do. I did happen to notice one other anachronism, but this one escaped me. I suspect we would find a lot of stale doc if we were to look through Drake carefully. :)
___
*[multibody/plant/multibody_plant.h, line 519 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-MuztGu9APZP7kl6oG2j:-Mv-GFktE4V1RZtcDmGH:b-lcdubc) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/multibody/plant/multibody_plant.h#L519)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

nit, maybe say: "between bodies welded (directly or transitively) to the world". Not sure if this as is implies that islands of welded bodies to the world are actually filtered or not. They are. Fell free to reword or ignore if clear to you.
</blockquote></details>

Will do. Suggested text is clearer, and may matter a bit more as I start to work on #16422.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: 1 unresolved discussion (waiting on @rpoyner-tri)

___
*[multibody/plant/multibody_plant.h, line 1164 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Mv-j_AS2x-kdbjSGX59:-Mv-j_AT9CNh9kgeO_Bk:bwblacz) ([raw file](https://github.com/robotlocomotion/drake/blob/20030a3a6879c7d40af368ad49769dea1cfb5eee/multibody/plant/multibody_plant.h#L1164)):*
> ```objc
>   /// If `this` plant registered geometry with a SceneGraph:
>   /// - input and output ports to enable communication with that SceneGraph are
>   ///   declared,
> ```

The SG-related ports are always declared whether or not any geometry was registered.  They also happen in the constructor, not Finalize.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: 1 unresolved discussion (waiting on @rpoyner-tri)

___
*[multibody/plant/multibody_plant.h, line 1164 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Mv-j_AS2x-kdbjSGX59:-Mv-zbbZ4CJ8_kB2Xbug:bvue2kz) ([raw file](https://github.com/robotlocomotion/drake/blob/20030a3a6879c7d40af368ad49769dea1cfb5eee/multibody/plant/multibody_plant.h#L1164)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

whee, more anachronisms. working. Do I need to just delete this lie, or restructure all mentions of SG ports?
</blockquote></details>

I don't need you to fix lines that you haven't already touched; just not re-itemizing this lie is enough for me.

Since there is a whole chapter about what Finalize does already hyperlinked here, I think any bullet list in the function docs that tries to summarize it is a lost cause.  I'd favor nuking the whole list (and possibly making the cross-reference more prominent).

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: 1 unresolved discussion (waiting on @jwnimmer-tri)

___
*[multibody/plant/multibody_plant.h, line 1164 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Mv-j_AS2x-kdbjSGX59:-Mv00cE03Lh9cq-vTWiH:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/20030a3a6879c7d40af368ad49769dea1cfb5eee/multibody/plant/multibody_plant.h#L1164)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

I don't need you to fix lines that you haven't already touched; just not re-itemizing this lie is enough for me.

Since there is a whole chapter about what Finalize does already hyperlinked here, I think any bullet list in the function docs that tries to summarize it is a lost cause.  I'd favor nuking the whole list (and possibly making the cross-reference more prominent).
</blockquote></details>

Done.

**Review by amcastro-tri — COMMENTED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: 1 unresolved discussion (waiting on @jwnimmer-tri)

___
*[geometry/scene_graph.h, line 811 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16490#-Muzpw2g7boouoRDXBMm:-Mv03Zw87k36mRARtkE0:bkqfkc9) ([raw file](https://github.com/robotlocomotion/drake/blob/76adfad8dbbf686c2ca419615f070ceaae86567f/geometry/scene_graph.h#L811)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

This is a good comment -- you are making me think harder. I did originally mean mbp or its clients, thinking really of "its clients" as application code, essentially. My assumption was that SceneGraph doesn't really have any other viable partners/consumers than mbp. I think that assumption is largely true, or at least, I don't know of any significant counterexamples.

But, as @ggould-tri teaches, "if you're reviewer is confused, your work is objectively confusing." Perhaps I'll work on a phrasing that is even less prescriptive about what code can make filtering decisions.
</blockquote></details>

Thanks! much more clear. 
Even though SG's biggetst client today is MBP, it was designed to be a completely separate entity. Therefore the door is absolutely open for other uses where only geometric queries are needed but not MBP is needed (e.g., a system with hand-written dynamics or even no dynamics at all but only prescribed motions.)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16490)* status: :shipit: complete! all discussions resolved, LGTM from assignees jwnimmer-tri(platform),amcastro-tri (waiting on @rpoyner-tri)
